### PR TITLE
Use SPIFFE ID for validation if enabled on current issuing and validating MRC certificate

### PR DESCRIPTION
**Pull request by jsturtevant**
Signed-off-by: James Sturtevant <jstur@microsoft.com>

**Description**:
This builds on #5131 which added the SPIFFE ID into the certificates.  Now that the SPIFFE ID is present we can use it to make routing and validation decisions.

Todo:

- [x] add e2e tests
- [x] add more unit tests

Fixes: #5025, #5026

**Testing done**:

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [x] |
| CI System                  | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [x] |
| Control Plane              | [x] |
| Demo                       | [ ] |
| Documentation              | [ ] |
| Egress                     | [ ] |
| Ingress                    | [ ] |
| Install                    | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| Performance                | [ ] |
| SMI Policy                 | [ ] |
| Security                   | [ ] |
| Sidecar Injection          | [ ] |
| Tests                      | [ ] |
| Upgrade                    | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?
no

2. Is this a breaking change?
no

4. Has documentation corresponding to this change been updated in the [osm-docs](https://github.com/openservicemesh/osm-docs) repo (if applicable)?
no

### CODE REVIEW

**Comment by jsturtevant** [comment]
I've opened this so can get feedback on this approach. 

I considered enlightening the `idenity.ServiceIdenitity` to be more than a string which would allow for  calls like `si.AsPrincipal(b.trustDomain)` to stay the same instead of giving this information to the builders to construct the appropriate rules but after trying it out this seemed like a cleaner approach and closer to what we have with the trust domain and other settings.  We might consider such a change to the `identity` package in the future but would encompass larger changes.

File: pkg/catalog/catalog.go
Comment (jsturtevant) [inline pkg/catalog/catalog.go:37]: going to make this more similar to `GetTrustDomain` which has less chances for various types of failures after #5036

**Comment by jsturtevant** [comment]
on hold until https://github.com/openservicemesh/osm/pull/5193 merges

File: pkg/certificate/manager.go
Comment (jsturtevant) [inline pkg/certificate/manager.go:149]: I kept this as a struct since it keeps it on the stack vs allocating a list in the heap based on https://github.com/openservicemesh/osm/pull/5193#discussion_r993584627

File: pkg/catalog/inbound_traffic_policies.go
Comment (jsturtevant) [reply pkg/catalog/inbound_traffic_policies.go]: https://github.com/openservicemesh/osm/issues/5236

File: pkg/catalog/catalog.go
Comment (jsturtevant) [reply pkg/catalog/catalog.go]: updated after #5193

**Comment by jsturtevant** [comment]
@openservicemesh/osm-maintainers this is ready for review 🚀

**Comment by jsturtevant** [comment]
rebased on all the latest changes

File: pkg/certificate/manager.go
Comment (jsturtevant) [reply pkg/certificate/manager.go:128]: No we don't know that information at this point.  Will update the comment

File: pkg/identity/types.go
Comment (jsturtevant) [reply pkg/identity/types.go:41]: I will double check, it is covered as part of the e2e tests

File: pkg/osm/callbacks.go
Comment (jsturtevant) [reply pkg/osm/callbacks.go:32]: The spiffe id will only be in the certificates once the signing issuer is issuing all the certificates.  Once that is the case we can switch to using the SPIFFE ID to gather the metadata about the proxy connecting instead of the Common name. 

When using SPIFFE as the standard, the identity in the certificate is the URI SAN not the common name.

This will be required once we are using SPIRE since SPIRE will create the SPIFFE ID in the URI field instead of common name.

Comment (jsturtevant) [reply pkg/osm/callbacks.go:32]: Should move this comment to below as well as update the line 30/31

Comment (jsturtevant) [reply pkg/osm/callbacks.go]: updated comment

**Review by keithmattix — APPROVED** [review]

**Review by jaellio — APPROVED** [review]

File: pkg/catalog/inbound_traffic_policies_test.go
Comment (jsturtevant) [reply pkg/catalog/inbound_traffic_policies_test.go:2370]: No, I think I lost them during a rebase.  Let me add that back in!

Comment (jsturtevant) [reply pkg/catalog/inbound_traffic_policies_test.go:2371]: added a test back in

**Pull request merged**
